src/App.js

```diff
@@ -1,20 +1,33 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = useState(
+    JSON.parse(localStorage.getItem("savedTodoList")) || []
+  );
+
+  useEffect(() => {
+    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+  }, [todoList]);
+
+  return [todoList, setTodoList];
+};
+
 function App() {
-  const [todoList, setTodoList] = useState([]);
+  const [todoList, setTodoList] = useSemiPersistentState();
 
   const addTodo = (newTodo) => {
     setTodoList([...todoList, newTodo]);
   };
```

Comment:
So this is my fault -- After more research, I found the way we set the new array was wrong. React allows us to provide a function to the setter (setTodoList) and function param will get the most recent value of the state.

So it'd look something like this:

``suggestion
  const addTodo = (newTodo) => {
     setTodoList( (previousTodoList) => [...previousTodoList, newTodo])
  };
```
I used an arrow function there b/c it's easier to read. How does all this sound to you?

@[kwilson05](https://github.com/kwilson05) kwilson05 [2 hours ago](https://github.com/omer-jan/react-cockatoo/pull/5#discussion_r1106529701)
Exercise 1.4 overview section actually explains my comment a bit better 😆
https://github.com/Code-the-Dream-School/react-cockatoo/wiki/Lesson-1.4
take a look

Reply:
Is this because we pass the copy of the array or object into the setter function?

Reply:
Doing it this gurrantees us getting the latest state according to React. Not sure why but they reccomend this way haha. 
Maybe the other way..there's a chance you get a stale/old state

Reply:
Okay, Thank you!🙂